opencog/atomutils/Unify.cc

```diff
@@ -348,14 +355,126 @@ bool Unify::is_bound_to_ancestor(const Variables& variables,
 
 Handle Unify::substitute(BindLinkPtr bl, const TypedSubstitution& ts)
 {
-	// Perform alpha-conversion, this will work over values that are
-	// non variables as well
-	//
 	// TODO: make sure that ts.second contains the declaration of all
 	// variables
-	Handle h = bl->alpha_conversion(strip_context(ts.first), ts.second);
+	return substitute(bl, strip_context(ts.first), ts.second);
+}
+
+Handle Unify::substitute(BindLinkPtr bl, const HandleMap& var2val,
+                         Handle vardecl)
```

Comment:
I initially put it in `BindLink`, but then I realized that it slightly collides in name with `LambdaLink::substitute` which is doing something a bit different. The former is a more specific kind of substitution that also removes constant clauses and useless/ill quotations and the unifier is the only consumer of it, so I thought it's better to move it to `Unify`. But maybe it could be in `BindLink`.